### PR TITLE
return aerohive serial

### PR DESCRIPTION
**Pull request by inphobia**
no idea why this is needed, but without this change no serial is returned:

netdisco-do show -d hiveap207 -e serial
[25474] 2018-11-09 19:45:09  info App::Netdisco version 2.039031 loaded.
[25474] 2018-11-09 19:45:10  info show: [10.40.252.14]/serial started at Fri Nov  9 20:45:10 2018
undef
[25474] 2018-11-09 19:45:10  info show: finished at Fri Nov  9 20:45:10 2018
[25474] 2018-11-09 19:45:10  info show: status done: Showed serial response from 10.40.252.14

netdisco-do show -d hiveap207 -e Layer2::Aerohive::serial
[25491] 2018-11-09 19:45:15  info App::Netdisco version 2.039031 loaded.
[25491] 2018-11-09 19:45:15  info show: [10.40.252.14]/Layer2::Aerohive::serial started at Fri Nov  9 20:45:15 2018
undef
[25491] 2018-11-09 19:45:16  info show: finished at Fri Nov  9 20:45:16 2018
[25491] 2018-11-09 19:45:16  info show: status done: Showed serial response from 10.40.252.14

with the change:
netdisco-do show -d hiveap207 -e Layer2::Aerohive::serial
[23072] 2018-11-09 19:44:06  info App::Netdisco version 2.039031 loaded.
[23072] 2018-11-09 19:44:07  info show: [10.40.252.14]/Layer2::Aerohive::serial started at Fri Nov  9 20:44:07 2018
02501610110274
[23072] 2018-11-09 19:44:07  info show: finished at Fri Nov  9 20:44:07 2018
[23072] 2018-11-09 19:44:07  info show: status done: Showed serial response from 10.40.252.14

netdisco-do show -d hiveap207 -e serial
[23075] 2018-11-09 19:44:13  info App::Netdisco version 2.039031 loaded.
[23075] 2018-11-09 19:44:13  info show: [10.40.252.14]/serial started at Fri Nov  9 20:44:13 2018
02501610110274
[23075] 2018-11-09 19:44:14  info show: finished at Fri Nov  9 20:44:14 2018
[23075] 2018-11-09 19:44:14  info show: status done: Showed serial response from 10.40.252.14

### CODE REVIEW

**Comment by JeroenvIS** [comment]
This could potentially cause issues for older HiveOS versions. I assume that the "serial" method worked correctly in the past.
Perhaps it would be cleaner to keep a call to the parent serial() method in the code; either as primary source of serial number, or as a fallback in case ahSystemSerial() isn't returned by a device.

**Comment by JeroenvIS** [comment]
Maybe my assumption about "serial" working on Aerohive on older versions doesn't hold true; generic source for serial() would be from Layer2 class, which tries to derive serial number from ENTITY-MIB with fallback to an ancient Cisco OID. Just using ahSystemSerial() would be cleaner.

**Comment by inphobia** [comment]
guess i should have mentioned that.

for me getting ahsystemserial or generic layer2 serial does not work on hiveos6.5r10, they just dont return that oid:

snmpwalk -m all -M '-./netdisco-mibs/aerohive/' -v 2c -c xxx hiveap130 .1.3.6.1.4.1.26928.1.2

AH-SYSTEM-MIB::ahCpuUtilization.0 = INTEGER: 16                                                                                                               AH-SYSTEM-MIB::ahMemUtilization.0 = INTEGER: 68                                                                                                               AH-SYSTEM-MIB::ahClientCount.0 = INTEGER: 3
--

our newer ap's running hiveos 8.4r4 do return it with this change however, but generic serial detection also does not work however.


AH-SYSTEM-MIB::ahSystemName.0 = STRING: HiveAP256                                                                                                             AH-SYSTEM-MIB::ahSystemDescription.0 = STRING: Product:AP250, HWVersion:01, SoftWareVersion:HiveOS 8.4r4 build-207528                                         AH-SYSTEM-MIB::ahCpuUtilization.0 = INTEGER: 3                                                                                                                AH-SYSTEM-MIB::ahMemUtilization.0 = INTEGER: 30                                                                                                               AH-SYSTEM-MIB::ahSystemSerial.0 = STRING: 02501609292170                                                                                                      AH-SYSTEM-MIB::ahDeviceMode.0 = STRING: AP250                                                                                                                 AH-SYSTEM-MIB::ahUpTime.0 = STRING: 1 weeks, 5 days, 21 hours, 8 minutes, 48 seconds                                                                          AH-SYSTEM-MIB::ahHwVersion.0 = STRING: 01                                                                                                                     AH-SYSTEM-MIB::ahClientCount.0 = INTEGER: 5                                                                                                                   AH-SYSTEM-MIB::ahEnvirmentTemp.0 = INTEGER: 48                                                                                                                AH-SYSTEM-MIB::ahEnvirmentFan.0 = INTEGER: 0                                                                                                                  AH-SYSTEM-MIB::ahFirmwareVersion.0 = STRING: HiveOS 8.4r4 build-207528
--


so bottom line is, with this change getting serials at least works in my experience with newer hiveos versions, but for older versions no new regressions seem to be introduced.

i no longer have access to older versions of hiveos atm to test with those.

**Comment by JeroenvIS** [comment]
Thanks @ollyg for merging :+1:  I was caught up in other tasks and forgot this thread
